kotlinlib/src/mill/kotlinlib/android/AndroidHiltSupport.scala

```diff
@@ -0,0 +1,345 @@
+package mill.kotlinlib.android
+
+import coursier.Repository
+import mill.api.{PathRef, Result}
+import mill.define.{Discover, ExternalModule, ModuleRef}
+import mill.javalib.android.AndroidSdkModule
+import mill.kotlinlib.DepSyntax
+import mill.kotlinlib.android.AndroidHiltSupport.HiltGeneratedClasses
+import mill.kotlinlib.ksp.KspModule
+import mill.kotlinlib.worker.api.KotlinWorkerTarget
+import mill.scalalib.{Dep, JvmWorkerModule}
+import mill.{T, Task}
+import os.Path
+
+import java.io.File
+
+/**
+ * Trait for mixing in AndroidAppKotlinModule to
+ * support the Hilt Dependency Injection framework for Android.
+ *
+ * It prepends compilation steps by using Kotlin Symbol Processing
+ * for pre-processing the Hilt annotations and generating the necessary classes for Hilt to work, then
+ * compiles all the sources together with a Java pre-processor step and finally a transform ASM step
+ * to achieve the compile time dependency injection!
+ */
+@mill.api.experimental
+trait AndroidHiltSupport extends KspModule with AndroidAppKotlinModule {
+
+  /**
+   * With Hilt support, sources are compiled as part of the
+   * Hilt compilation flow which involves an asm transform step
+   * on the compiled sources, so we don't want to recompile them.
+   */
+  override def allSources: T[Seq[PathRef]] = generatedSources()
+
+  /**
+   * In the case of hilt, we are not adding any generated
+   * KSP files in generated sources - yet
+   */
+  override def generatedSources: T[Seq[PathRef]] =
+    super[AndroidAppKotlinModule].generatedSources()
```

Comment:
Do we really need to specify a specific `super` here? Can you add a comment what issue this solves. Why isn't a normal `super.generatedSources()` enough?

Reply:
this is resolved after the latest simplification